### PR TITLE
Add general usage page, improve guides navigation and content

### PR DESCRIPTION
**Pull request by marga-kinvolk**
This PR includes three documentation changes:
- Add a guide documenting common parameters (pod selection, output formatting, kubernetes runtimes CLI)
- Review all guides to ensure that the gadgets are named correctly and also fix typos and so on.
- Add descriptions to all guides, for the Overview page. Which looks like this:
image

Fixes: #495

### CODE REVIEW

File: docs/guides/general-usage.md
Comment (mauriciovasquezbernal) [inline docs/guides/general-usage.md:21]: nit: lower case after `,`?: Show -> show

File: docs/guides/general-usage.md
Comment (mauriciovasquezbernal) [inline docs/guides/general-usage.md:43]: output -> Output?

File: docs/guides/advise/network-policy.md
Comment (mauriciovasquezbernal) [inline docs/guides/advise/network-policy.md:5]: Can we make it an exact match of https://github.com/kinvolk/inspektor-gadget/blob/main/cmd/kubectl-gadget/advise/network-policy.go#L40? Same for others.

File: docs/guides/top/block-io.md
Comment (mauriciovasquezbernal) [inline docs/guides/top/block-io.md:8]: nit: in other guides the gadgets name is not surrounded by '``'

**Review by mauriciovasquezbernal — COMMENTED** [review]
Some minor nits. Do you think this can live together with https://github.com/kinvolk/inspektor-gadget/pull/651?

File: docs/guides/general-usage.md
Comment (marga-kinvolk) [reply docs/guides/general-usage.md]: Fixed, thanks

File: docs/guides/advise/network-policy.md
Comment (marga-kinvolk) [reply docs/guides/advise/network-policy.md]: Alright, I've made all descriptions match the help tags. In three cases (the profile gadgets and snapshot socket), I modified the help tags to match the description I had written. I think it's good that these match, but it bothers me that there's so many places that we need to manually sync (there's also the README, which I also updated for my three changes). I wonder if there's a better way of doing that...

File: docs/guides/top/block-io.md
Comment (marga-kinvolk) [reply docs/guides/top/block-io.md]: Yeah, the original guides were not consistent in this sense. I've removed the inverted quotes around the names of the gadgets throughout.

**Comment by marga-kinvolk** [comment]
Sorry for the delay here. As mentioned yesterday, I was waiting for your other change to be merged, to avoid conflicts. I think this should be ready to merge now, so if you're happy with my changes feel free to merge it yourself.

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM, merging!

**Pull request merged**
